src/Aspire.Cli/DotNetCliRunner.cs

```diff
@@ -530,10 +558,16 @@ private async Task StartBackchannelAsync(Process process, string socketPath, Tas
                 logger.LogTrace("Attempting to connect to AppHost backchannel at {SocketPath} (attempt {Attempt})", socketPath, connectionAttempts++);
                 await backchannel.ConnectAsync(socketPath, cancellationToken).ConfigureAwait(false);
                 backchannelCompletionSource.SetResult(backchannel);
+                backchannel.AddDisconnectHandler((_, _) =>
+                {
+                    // If the backchannel disconnects, we want to stop the CLI process
+                    Environment.Exit(ExitCodeConstants.Success);
```

Comment:
This also affects non-extension cli runs @mitchdenny 

I'm assuming that the CLI connection is non-recoverable and we can assume that a disconnect means that the AppHost has exited (and thus we can exit the cli?)